helm/cluster/README.md

```diff
@@ -437,12 +451,14 @@ Provider-specific properties that can be set by cluster-$provider chart in order
 | `providerIntegration.resourcesApi.clusterResourceEnabled` | **Cluster resource enabled** - Flag that indicates if the Cluster resource is enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `true`|
 | `providerIntegration.resourcesApi.controlPlaneResourceEnabled` | **Control plane resource enabled** - Flag that indicates if the control plane resource is enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `true`|
+| `providerIntegration.resourcesApi.helmRepositoryResourcesEnabled` | **HelmRepository resources enabled** - Flag that indicates if the HelmRepository resources are enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `false`|
 | `providerIntegration.resourcesApi.infrastructureCluster` | **Infrastructure cluster** - Group, version and kind of provider-specific infrastructure cluster resource.|**Type:** `object`<br/>|
 | `providerIntegration.resourcesApi.infrastructureCluster.group` | **API group**|**Type:** `string`<br/>**Example:** `"infrastructure.cluster.x-k8s.io"`<br/>|
 | `providerIntegration.resourcesApi.infrastructureCluster.kind` | **API kind**|**Type:** `string`<br/>**Examples:** `"AWSCluster", "AzureCluster", "VCDCluster", "VSphereCluster"`<br/>|
 | `providerIntegration.resourcesApi.infrastructureCluster.version` | **API version**|**Type:** `string`<br/>**Examples:** `"v1alpha1", "v1beta1", "v1beta2", "v1", "v2"`<br/>|
 | `providerIntegration.resourcesApi.machineHealthCheckResourceEnabled` | **MachineHealthCheck resource enabled** - Flag that indicates if the MachineHealthCheck resource is enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `true`|
 | `providerIntegration.resourcesApi.machinePoolResourcesEnabled` | **Machine pool resources enabled** - Flag that indicates if the machine pool resources are enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `true`|
+| `providerIntegration.resourcesApi.verticalPodAutoscalerCrdHelmReleaseResourceEnabled` | **vertical-pod-autoscaler-crd HelmRelease resource enabled** - Flag that indicates if the vertical-pod-autoscaler-crd HelmRelease is enabled and templated. This is meant only for the initial development purposes for the sake of incrementally integrating cluster chart into cluster-$provider apps.|**Type:** `boolean`<br/>**Default:** `false`|
```

Comment:
just an idea instead of adding each component via flag as `providerIntegration.resourcesApi.verticalPodAutoscalerCrdHelmReleaseResourceEnabled`

we could try to add them all under one roof
e.g. 

providerIntegration.resourcesApi.helmReleaseResources.


- verticalPodAutoscalerCrdHelmReleaseResourceEnabled
- coreDNS...
- cilium ...
- ...

Reply:
I wanted to do that first, but then went with per-HelmRelease flag, so every PR in cluster chart can be merged independently, and that we can introduce HelmReleases into cluster-\<provider\> apps more iteratively one-by-one if needed (e.g. VPA CRD works, but there are issues with cilium, so we enable VPA CRD, but disable cilium).

Also there slight differences across providers, e.g. one provider using some HelmRelease, another not.